api/observability/v1/output_types.go

```diff
@@ -593,12 +593,17 @@ type HTTP struct {
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="Timeout",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:number"}
 	Timeout int `json:"timeout,omitempty"`
 
-	// Method specifies the Http method to be used for sending logs. If not set, 'POST' is used.
+	// Method specifies the HTTP method to be used for sending logs. If not set, 'POST' is used.
 	//
 	// +kubebuilder:validation:Enum:=GET;HEAD;POST;PUT;DELETE;OPTIONS;TRACE;PATCH
 	// +kubebuilder:validation:Optional
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="HTTP Method"
 	Method string `json:"method,omitempty"`
+
+	// ProxyURL if specified is used to proxy HTTP and HTTPS traffic.
+	//
+	// +kubebuilder:validation:Optional
+	ProxyURL string `json:"proxyURL,omitempty"`
```

Comment:
@alanconway What is our motivation for not configuring all three options?  I have some thoughts why you designed it this way but would like to understand if they are the same as mine

- Needs URL validation: https://github.com/openshift/cluster-logging-operator/pull/2798/files#diff-4749b2553f93d0fa25f19c74b7fad9dc691f10c0bf9c8f4b46ab0681c4a9560dR663

Reply:
- enabled: redundant - if proxy config exists, it's enabled if not it's disabled.
- http vs. https: if user has different proxies for each they can set up different outputs.

You could argue that configuring http/https separately is more convenient, and for the general web proxy case it probably is. However in the log forwarding case, each output will either be secure or insecure based on the TLS configuration, there's no case where we will send both to the same output. So this PR enable the proxy proxy config to handle anything, and lets the TLS config determine whether we send secure or insecure.

Reply:
added url valiation